packages/react-native-client/src/hooks/usePeers.ts

```diff
@@ -1,13 +1,90 @@
 import { useEffect, useState } from 'react';
 
-import {
-  Endpoint,
-  EndpointsUpdateEvent,
-  Metadata,
-} from '../RNFishjamClient.types';
-import RNFishjamClientModule from '../RNFishjamClientModule';
+import { Metadata, SimulcastConfig, TrackEncoding } from '../types';
+import RNFishjamClientModule from '../FishjamClient';
 import { ReceivableEvents, eventEmitter } from '../common/eventEmitter';
 
+export enum TrackType {
+  Audio = 'Audio',
+  Video = 'Video',
+}
+
+/**
+ * Type describing Voice Activity Detection statuses.
+ *
+ * SPEECH - voice activity has been detected
+ * SILENCE - lack of voice activity has been detected
+ */
+export enum VadStatus {
+  Silence = 'silence',
+  Speech = 'speech',
+}
+
+export type Track<MetadataType extends Metadata> = {
+  id: string;
+  type: TrackType;
+  metadata: MetadataType;
+  vadStatus: VadStatus;
+  // Encoding that is currently received. Only present for remote tracks.
+  encoding: TrackEncoding | null;
+  // Information about simulcast, if null simulcast is not enabled
+  simulcastConfig: SimulcastConfig | null;
+  // The reason of currently selected encoding. Only present for remote tracks.
+  encodingReason: EncodingReason | null;
+};
+
+/**
+ * Type describing possible reasons of currently selected encoding.
+ *
+ * - OTHER - the exact reason couldn't be determined
+ * - ENCODING_INACTIVE - previously selected encoding became inactive
+ * - LOW_BANDWIDTH - there is no longer enough bandwidth to maintain previously selected encoding
+ */
+export enum EncodingReason {
+  Other = 'other',
+  EncodingInactive = 'encodingInactive',
+  LowBandwidth = 'lowBandwidth',
+}
+
+export type EndpointsUpdateEvent<
```

Comment:
Rename to `PeersUpdateEvent` ?

Reply:
I wanted to make this rename in separate PR. But maybe it make sense to do it now

Reply:
I was thinking about it - but to update Endpoint to Peers everywhere, i would require more changes (also in native code, due to events). There is another ticket to do this ([FCE-212](https://linear.app/swmansion/issue/FCE-212/removerename-references-to-endpoint-from-rn-sdk-api)), so let's leave old names for now - and update them with new task.

Reply:
👍